arch/arm/core/aarch64/fatal.c

```diff
@@ -163,12 +163,13 @@ void z_arm64_fatal_error(unsigned int reason, const z_arch_esf_t *esf)
 		__asm__ volatile("mrs %0, CurrentEL" : "=r" (el));
 
 		if (GET_EL(el) != MODE_EL0) {
-			__asm__ volatile("mrs %0, esr_el1" : "=r" (esr));
-			__asm__ volatile("mrs %0, far_el1" : "=r" (far));
-			__asm__ volatile("mrs %0, elr_el1" : "=r" (elr));
-
+			esr = read_sysreg(esr_el1);
```

Comment:
I meant replacing __asm__ volatile everywhere in arm64 code with read_sysreg/write_sysreg, not just this file. Can take it up later as improvement if not now.

Reply:
hooo ok. I'm preparing a PR just for cleanup stuff, I'll add the commit in that PR. For this PR I think this is enough.

Reply:
In that case, I recommend removing this commit from current PR and taking care of it in a single commit in next PR.

Reply:
ACK. Removing it.